mesop/components/autocomplete/autocomplete.py

```diff
@@ -0,0 +1,229 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Iterable, Literal
+
+import mesop.components.autocomplete.autocomplete_pb2 as autocomplete_pb
+from mesop.component_helpers import (
+  Style,
+  insert_component,
+  register_event_handler,
+  register_event_mapper,
+  register_native_component,
+)
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class AutocompleteOption:
+  """
+  Represents an option in the autocomplete drop down.
+
+  Attributes:
+    label: Content to show for the autocomplete option
+    value: The value of this autocomplete option.
+  """
+
+  label: str | None = None
+  value: str | None = None
+
+
+@dataclass(kw_only=True)
+class AutocompleteOptionGroup:
+  """
+  Represents an option group to group options in the autocomplete drop down.
+
+  Attributes:
+    label: Group label
+    options: Autocomplete options under this group
+  """
+
+  label: str
+  options: list[AutocompleteOption]
+
+
+@dataclass(kw_only=True)
+class AutocompleteInputEvent(MesopEvent):
+  """Represents an on input event on an autocomplete component.
+
+  This event should mainly be used custom filtering of autocomplate results based on
+  the given user input.
+
+  Attributes:
+    value: Input value.
+    key (str): key of the component that emitted this event.
+  """
+
+  value: str
```

Comment:
I think it's OK to use the InputEvent already defined. https://github.com/google/mesop/blob/c9e38cdf024e7ee175f2fe9fe2521bf938ff76b0/mesop/events/events.py#L22

If we need to add more properties for autocomplete input specifically, we could define AutocompleteInputEvent as a subclass of InputEvent

Reply:
Ah I see. Didn't realize I could reuse the events. Updated